include/aws/common/assert.h

```diff
@@ -33,6 +32,21 @@ void aws_fatal_assert(const char *cond_str, const char *file, int line) AWS_ATTR
 AWS_COMMON_API
 void aws_backtrace_print(FILE *fp, void *call_site_data);
 
+typedef void (*assert_handler_fn_t)(const char *cond_str, const char *file, int line);
```

Comment:
2 things here:
* prefix the name with aws_
* Make the typedef not a pointer type, and put the pointer on the parameter.